README.md

```diff
@@ -711,6 +711,19 @@ end
 
 ```
 
+You may wish to have elements declared in a page object class that are not always guaranteed to be present (success or error messages, etc.). If you'd still like to test such a page with `all_there?` you can define an `excluded_elements` method on your page object class that excludes the specified elements from the `all_there?` check.
+
+```ruby
+class TestPage < SitePrism::Page
+  element(:name_field, '#name')
+  element(:success_msg, 'span.alert-success')
+
+  def excluded_elements
```

Comment:
Why not just have this as an array?  Why a method?

Reply:
I'm not sure I understand the question - do you mean why a method instead of a class constant or an instance variable? Given that the logic I added in lib/site_prism/element_checker.rb is meant to operate on `excluded_elements` as some kind of member of a page object class, I'm not sure what having this be a simple array would mean. It has to be either some kind of member variable that holds an array, or what I have here, a method that returns an array.

Reply:
What I'm trying to say is that this is the only point in an otherwise declarative DSL where the user is instructed that they need to define a method, so it maybe feels a little out of place.


```ruby
class TestPage < SitePrism::Page
  element(:name_field, '#name')
  element(:success_msg, 'span.alert-success')

  def excluded_elements
    %w[success_msg]
  end
end
```

I'm wondering if you considered something like the below implementation:

```ruby
class TestPage < SitePrism::Page
  element(:name_field, '#name')
  element(:success_msg, 'span.alert-success')

  excluded_elements ['success_msg', 'alert']
 
end
```

Reply:
Gotcha. Yeah, that is a fair point, keeping up with the declarative DSL probably is going to flow a bit better. I'll give that a spin. Thanks!